builtin/logical/pki/acme_state.go

```diff
@@ -186,6 +187,50 @@ func (a *acmeState) LoadJWK(ac *acmeContext, keyID string) ([]byte, error) {
 	return key.Jwk, nil
 }
 
+func (a *acmeState) LoadAuthorization(ac *acmeContext, userCtx *jwsCtx, authId string) (*ACMEAuthorization, error) {
+	if authId == "" {
+		return nil, fmt.Errorf("malformed authorization identifier")
+	}
+
+	entry, err := ac.sc.Storage.Get(ac.sc.Context, acmeAuthroizationPrefix+authId)
+	if err != nil {
+		return nil, fmt.Errorf("error loading authorization: %w", err)
+	}
+
+	if entry == nil {
+		return nil, fmt.Errorf("authorization does not exist: %w", ErrMalformed)
+	}
+
+	var authz ACMEAuthorization
+	err = entry.DecodeJSON(&authz)
+	if err != nil {
+		return nil, fmt.Errorf("error decoding authorization: %w", err)
+	}
+
+	if userCtx.Kid != authz.AccountId {
+		return nil, ErrUnauthorized
+	}
+
+	return &authz, nil
+}
+
+func (a *acmeState) SaveAuthorization(ac *acmeContext, authz *ACMEAuthorization) error {
+	if authz.Id == "" {
+		authz.Id = genUuid()
```

Comment:
nit: I know this will most likely change but I find this is a tad odd that genUuid is called all the way down here but we never return it in any way?

Reply:
Since `authz` is a pointer receiver, I think we'd get it for free there in the return...?

Reply:
It would mean you'd need to persist your authz object before you persist your order object on creation, but I don't think that's too out of the ordinary, you'd prefer dangling authz to orders without authz.

Reply:
I suppose, I'd personally rather just see it generated when we construct a new authz vs loading a new one. Anyways nothing that is preventing this from merging now.

Reply:
I'm confused, isn't this what'd be called on constructing a new Authz? This doesn't load the authz, just persists changes back to storage.

Reply:
Well this doesn't construct a new `ACMEAuthorization` either? Right now nothing seems to be creating and properly filling out an initial `ACMEAuthorization` object, not an existing one being loaded from storage, but if that did exist, that is where I'd expect to see the call to getUuid() not here within the save step.

Reply:
Right, I was envisioning the order would create the new ACMEAuthorization(s), call save on them, then call save on the order (also generating its ID), and then return the blob to the caller. My 2c tho :-)

Reply:
Ah, I see, importIssuer creates the issuerId, but the save call assumes an id. We can remove this and have a separate create call and save call.